src/bin/cargo-info/command/info.rs

```diff
@@ -91,9 +97,13 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         &config_args,
     )?;
 
-    let pkgid = args.get_one::<String>("pkgid").map(String::as_str).unwrap();
+    let package = args
+        .get_one::<String>("package")
+        .map(String::as_str)
+        .unwrap();
+    let spec = PackageIdSpec::parse(package)?;
```

Comment:
Do you think we should check if the URL spec passed?

Reply:
We should either respect it or error if its there.  This is where reusing `PackageIdSpec` can be tricky.

Reply:
With how we are doing filtering, it looks like we should be respecting it and can likely leave it along.